File: solutions/scala/problem-00016.scala
/*
[minhuang@mocha:scala]$ time scala problem-00016.scala 
1366

real  0m1.362s
user  0m0.228s
sys 0m0.056s
*/
import scala.Math._

val n = BigInt(2).pow(1000)
println(n.toString.map(s => BigInt(s.toString)).reduceLeft[BigInt](_+_))

File: solutions/ruby/problem-00016.rb
#!/usr/bin/env ruby
# [minhuang@mocha:ruby]$ time ruby problem-00016.rb 
# 1366
#
# real  0m0.021s
# user  0m0.012s
# sys 0m0.012s
puts (2 ** 1000).to_s.chars.map { |c| c.to_i }.inject(:+)
